src/Symfony/Component/Messenger/Transport/TransportFactory.php

```diff
@@ -66,4 +71,39 @@ public function supports(#[\SensitiveParameter] string $dsn, array $options): bo
 
         return false;
     }
+
+    private function santitizeDsn(string $dsn): ?string
+    {
+        $parts = parse_url($dsn);
+        $dsn = '';
+
+        if (isset($parts['scheme'])) {
+            $dsn .= $parts['scheme'].'://';
+        }
+
+        if (isset($parts['user'])) {
+            $dsn .= $parts['user'];
```

Comment:
I'm not sure about this: there are schemes where there's no user:pass but just a token, and PHP will confuse that with the user, leading to a leak here

Reply:
hm, what would you suggest? unconditionally blank out the user or only blank out the user if the pass is not present?

Reply:
`isset($parts['user'] || isset($parts['pass']) ? '******@' : ''`

Reply:
fixed